### DISCUSSION TITLE
Install the Earth Analytics Python environment on your computer

### DISCUSSION THREAD

**Discussion by eculler** [earthlab-education/Earth-Analytics-2023-01-Intro#589] [Technical Demo Videos]
@earthlab-education/ea-students @earthlab-education/ea-instructors 

This is what we're doing in class this week. It should give you an alternative to Codespaces in case you're having performance issues.

Here are the instructions for installing your environment: https://earthlab-education.github.io/Earth-Analytics-2023-01-Intro/python-open-source-tool-setup/setup-04-setup-conda-earth-analytics-environment.html

In addition, you should:

1. (optional) Install [GitHub Desktop](https://desktop.github.com/) rather than working with the command line
2. Switch to the `upgrade-to-3.9` branch in the `earth-analytics-python-env` repository BEFORE running `mamba env create -f environment.yml`
3. Once your environment is installed and activated, get the development version of `earthpy` (with the new APPEEARS and M2M code!) using the command `pip install git+https://github.com/earthlab/earthpy@apppears`

Here's a video using an M1 Mac (thanks Advyth!): https://cuboulder.zoom.us/rec/share/uquUnF-kX0QtnOkKHauMCDN2kUp0vl4MkV80og0rPvGaK7A36papNP1AqdwfiH0P.MXNSkMFVVGioRGjV

**Comment by misterskye** [comment]
Good evening everyone,
Not sure why, but getting the error below when I try to pip install earthpy@apppears.

Thanks

scorbett@IGSWMBWGLTGM941 MINGW64 ~/earth_analytics/earth-analytics-python-env (upgrade-to-3.9)
$ pip install git+https://github.com/earthlab/earthpy@apppears
Collecting git+https://github.com/earthlab/earthpy@apppears
  Cloning https://github.com/earthlab/earthpy (to revision apppears) to c:\users\scorbett\appdata\local\temp\4\pip-req-build-pnuba326
  Running command git clone --filter=blob:none --quiet https://github.com/earthlab/earthpy 'C:\Users\scorbett\AppData\Local\Temp\4\pip-req-build-pnuba326'
  Running command git checkout -b apppears --track origin/apppears
  branch 'apppears' set up to track 'origin/apppears'.
  Switched to a new branch 'apppears'
  Resolved https://github.com/earthlab/earthpy to commit f9db8be772c6b6f06a43d036f96069c919f6f622
  Installing build dependencies ... error
  error: subprocess-exited-with-error

  × pip subprocess to install build dependencies did not run successfully.
  │ exit code: 1
  ╰─> [9 lines of output]
      WARNING: Retrying (Retry(total=4, connect=None, read=None, redirect=None, status=None)) after connection broken by 'SSLError(SSLCertVerificationError(1, '[SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:1007)'))': /simple/setuptools/
      WARNING: Retrying (Retry(total=3, connect=None, read=None, redirect=None, status=None)) after connection broken by 'SSLError(SSLCertVerificationError(1, '[SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:1007)'))': /simple/setuptools/
      WARNING: Retrying (Retry(total=2, connect=None, read=None, redirect=None, status=None)) after connection broken by 'SSLError(SSLCertVerificationError(1, '[SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:1007)'))': /simple/setuptools/
      WARNING: Retrying (Retry(total=1, connect=None, read=None, redirect=None, status=None)) after connection broken by 'SSLError(SSLCertVerificationError(1, '[SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:1007)'))': /simple/setuptools/
      WARNING: Retrying (Retry(total=0, connect=None, read=None, redirect=None, status=None)) after connection broken by 'SSLError(SSLCertVerificationError(1, '[SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:1007)'))': /simple/setuptools/
      Could not fetch URL https://pypi.org/simple/setuptools/: There was a problem confirming the ssl certificate: HTTPSConnectionPool(host='pypi.org', port=443): Max retries exceeded with url: /simple/setuptools/ (Caused by SSLError(SSLCertVerificationError(1, '[SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:1007)'))) - skipping
      ERROR: Could not find a version that satisfies the requirement setuptools>=40.8.0 (from versions: none)
      ERROR: No matching distribution found for setuptools>=40.8.0
      Could not fetch URL https://pypi.org/simple/pip/: There was a problem confirming the ssl certificate: HTTPSConnectionPool(host='pypi.org', port=443): Max retries exceeded with url: /simple/pip/ (Caused by SSLError(SSLCertVerificationError(1, '[SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:1007)'))) - skipping
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
error: subprocess-exited-with-error

× pip subprocess to install build dependencies did not run successfully.
│ exit code: 1
╰─> See above for output.

note: This error originates from a subprocess, and is likely not a problem with pip.
Could not fetch URL https://pypi.org/simple/pip/: There was a problem confirming the ssl certificate: HTTPSConnectionPool(host='pypi.org', port=443): Max retries exceeded with url: /simple/pip/ (Caused by SSLError(SSLCertVerificationError(1, '[SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:1007)'))) - skipping
(earth-analytics-python)
scorbett@IGSWMBWGLTGM941 MINGW64 ~/earth_analytics/earth-analytics-python-env (upgrade-to-3.9)

**Comment by Adriannaluger** [comment]
@nquarder Is this something we could go over in class?? I am having troubles with it...

**Reply by nquarder** [reply]
Hey @Adriannaluger!! Sorry for the delay here. The plan for this week is to not have discussion again as we're hosting a hackathon event that is going to conflict w/ class time. Are you able to attend any office hours this week? I'm cc'ing @eculler here too as office hours times may shift around this week. Let's find a time to meet to help troubleshoot!

**Reply by Adriannaluger** [reply]
Is there going to be an instructional video for windows? @nquarder

**Reply by nquarder** [reply]
We do not have one of those currently! Would you be able to meet w/ @eculler and the two of you could go through the steps and record?

**Comment by jsmi374** [comment]
Hello @eculler @nquarder .... I am having trouble installing the Earth Analytics Python Environment on my computer. I made progress on this earlier in the semester, but it doesn't seem to be working. I sent an email to the both of you .... perhaps we can set up a meeting soon to trouble shoot.

**Comment by ellenalamont17** [comment]
I too am having issues getting the assignments into my local environment. I worked through setting up GitDesktop in the last class but I cannot seem to get the local copies of the homework into that environment.
